python/python/bystro/proteomics/scripts/get_uniprot_id_gene_name_mapping.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+"""Download a list of uniprot_id / gene_name pairs from Uniprot and store the result on disk as csv."""
+
+import pandas as pd
+import tqdm
+from unipressed import UniprotkbClient  # type: ignore[import]
+
+from bystro.utils.config import BYSTRO_PROJECT_ROOT
+
+
+def get_uniprot_id_gene_name_mapping() -> pd.DataFrame:
+    """Get a dataframe associating each uniprot id with its cognate gene names."""
+    APPROXIMATE_TOTAL_RECORDS = 240_000
+    query_results = UniprotkbClient.search(
+        query="(organism_name:homo)",
+    )
+    uniprot_id_gene_name_mapping = []
+    for record in tqdm.trange(query_results.each_record(), total=APPROXIMATE_TOTAL_RECORDS):
+        genes = record.get("genes", [])  # type: ignore[attr-defined]
+        gene_names = []
+        for gene in genes:
+            if (gene_name := gene.get("geneName")) and (value := gene_name.get("value")):
+                gene_names.append(value)
+        if not gene_names:
+            gene_names = [None]
```

Comment:
I think since our missing convention is '!' in the annotation, it makes more sense to go with that. I'm not sure if we longer-term want to necessarily not bring in regions of the genome that don't have a gene name in Swiss-Prot but do have a TrEMBL uniprot ID to the joined data in case some of the top fold changes for the proteomics are in that category and users still want to see the variants specified to that region. For the current version it makes the most sense to drop them. I'm making a note here that we may want to include whether something is in Swiss-Prot or TrEMBL as a separate column in future for those kinds of cases (Example UniprotID: D3WYY1).

Reply:
The mapping csv isn't intended to be user-facing, but only accessed through the pair of public functions `get_uniprot_ids_from_gene_name` and `get_gene_names_from_uniprot_id`.  Pandas natively understands conventional null values like `None` or `np.nan` to represent missing data, but doesn't understand `"!"` that way.  My recommendation would be to continue to use pythonic representations of missing data except where necessary to preserve backwards compatibility in user-facing data, wdyt?

Happy to drop the uniprot_ids without corresponding gene names entirely, though.

Reply:
Ok got it, as long as they don't interact with any potential querying (i.e. clinvar gets connected to this at some point and a record with Ubiquinone gets pulled if you were trying to query where the gene name is 'None') and it's not user-facing, I'm ok with that. May be safer to go with np.nan just in case.

I do think for now we should just drop uniprot_ids without corresponding gene names in Swiss-Prot entirely with a todo to add that information to a future prot tutorial for users unless we decide to add TrEMBL at some point and mark those accordingly.